src/app/@ansyn/ansyn/app-providers/overlay-source-providers/tb/tb-source-provider.ts

```diff
@@ -46,20 +55,67 @@ export class TBSourceProvider extends BaseOverlaySourceProvider {
 
 	fetch(fetchParams: IFetchParams): Observable<any> {
 		let geometry;
+		let queries = [];
 
 		if (fetchParams.region.type === 'Point') {
 			geometry = getPolygonByPointAndRadius((<any>fetchParams.region).coordinates).geometry as GeoJSON.Polygon;
 		} else {
 			geometry = fetchParams.region;
 		}
 
+		if (fetchParams.dataInputFilters.length !== 0) {
+			let sensorTypeMatch = true;
+			let sensorNameMatch = true;
+			let sensorTypes;
+			let sensorNames;
+
+			if (fetchParams.dataInputFilters[0].sensorType) {
+				sensorTypes = fetchParams.dataInputFilters.map(filter => filter.sensorType);
+				if (sensorTypes.find(sensorType => sensorType === 'others')) {
+					sensorTypeMatch = false;
+					sensorTypes = sensorTypes.filter(sensorType => !(sensorType === 'others'));
+				}
+			} else {
+				sensorTypes = [];
+			}
+
+			if (fetchParams.dataInputFilters[0].sensorName) {
+				sensorNames = fetchParams.dataInputFilters.map(filter => filter.sensorName);
+				if (sensorNames.find(sensorName => sensorName === 'others')) {
+					sensorNameMatch = false;
+					sensorNames = sensorNames.filter(sensorName => !(sensorName === 'others'));
+				}
+			} else {
+				sensorNames = [];
+			}
+
+			if (sensorTypes.length !== 0) {
+				queries.push({
+					field: 'inputData.sensor.type',
+					values: sensorTypes,
+					isMatch: sensorTypeMatch
+				})
+			}
+
+			if (sensorNames.length !== 0) {
+				queries.push({
+					field: 'inputData.sensor.name',
+					values: sensorNames,
+					isMatch: sensorNameMatch
+				})
+			}
+		}
+
+		console.log(`TB fetch body queries: ${JSON.stringify(queries)}`);
```

Comment:
Remove console.log

Reply:
I'll remove it after I'll check that it works OK